torch/_dynamo/symbolic_convert.py

```diff
@@ -1946,6 +1947,9 @@ def __init__(
         # as soon as we create the tracing context we should keep it active, so any calls
         # into dynamo apis can rely on finding it
         with tracing(self.output.tracing_context):
+            if _is_train_step_compiler(compiler_fn):
```

Comment:
Can we do this in a way that does not have `_is_train_step_compiler` in dynamo? this was the feedback in an earlier PR. https://github.com/pytorch/pytorch/pull/98461/

Reply:
i don't know what you mean exactly.
1) you don't want any logic inside dynamo checking if we're in train step mode
2) its ok to have logic for train step mode, but you prefer it to take a different form (like a bool that comes from the user's api call)

I initially wrote it as (2) and hated how many lines of code i had to plumb "train_step=train_step" through to achieve this.  My current approach makes the change in just one place where it hands off from detecting the compile_fn type to marking the tracing context.

Reply:
> you don't want any logic inside dynamo checking if we're in train step mode

Correct - dynamo should not know bout train step compiler

Reply:
but... am I wrong to think that dynamo _has_ to know about train step compiler in order to decide to not graph-break on backwards?